solidity/core/test/validator-manager/multisigValidatorManager.test.ts

```diff
@@ -0,0 +1,264 @@
+    });
+
+    it('reverts if the new quorum threshold is zero', async () => {
+      await expect(validatorManager.setQuorumThreshold(0)).to.be.revertedWith(
+        '!range',
+      );
+    });
+
+    it('reverts if the new quorum threshold is greater than the validator set size', async () => {
+      await expect(validatorManager.setQuorumThreshold(3)).to.be.revertedWith(
+        '!range',
+      );
+    });
+
+    it('reverts when called by a non-owner', async () => {
+      await expect(
+        validatorManager.connect(nonOwner).setQuorumThreshold(2),
+      ).to.be.revertedWith('Ownable: caller is not the owner');
+    });
+  });
+
+  describe('#isQuorum', () => {
+    const root = ethers.utils.formatBytes32String('test root');
+    const index = 1;
+
+    beforeEach(async () => {
+      // Have 3 validators and a quorum of 2
+      await validatorManager.enrollValidator(validator1.address);
+      await validatorManager.enrollValidator(validator2.address);
+
+      await validatorManager.setQuorumThreshold(2);
+    });
+
+    it('returns true when there is a quorum', async () => {
+      const signatures = await signCheckpoint(root, index, [
+        validator0,
+        validator1,
+      ]);
+      expect(await validatorManager.isQuorum(root, index, signatures)).to.be
+        .true;
+    });
+
+    it('returns true when a quorum exists even if provided with non-validator signatures', async () => {
+      const signatures = await signCheckpoint(
+        root,
+        index,
+        [validator0, validator1, validator3], // validator 3 is not enrolled
+      );
+      expect(await validatorManager.isQuorum(root, index, signatures)).to.be
+        .true;
+    });
+
+    it('returns false when the signature count is less than the quorum threshold', async () => {
+      const signatures = await signCheckpoint(root, index, [validator0]);
+      expect(await validatorManager.isQuorum(root, index, signatures)).to.be
+        .false;
+    });
+
+    it('returns false when some signatures are not from enrolled validators', async () => {
+      const signatures = await signCheckpoint(
+        root,
+        index,
+        [validator0, validator3], // validator 3 is not enrolled
+      );
+      expect(await validatorManager.isQuorum(root, index, signatures)).to.be
+        .false;
+    });
+
+    it('reverts when signatures are not ordered by their signer', async () => {
+      // Reverse the signature order, purposely messing up the
+      // ascending sort
+      const signatures = (
+        await signCheckpoint(root, index, [validator0, validator1])
+      ).reverse();
+
+      await expect(
+        validatorManager.isQuorum(root, index, signatures),
+      ).to.be.revertedWith('!sorted signers');
+    });
+  });
+
+  describe('#_domainHash', () => {
+    it('matches Rust-produced domain hashes', async () => {
+      // Compare Rust output in json file to solidity output (json file matches
+      // hash for local domain of 1000)
```

Comment:
I suppose this is more important but can we not easily repro in javascript?

Reply:
yeah imo the merit of testing of these vectors is questionable to begin with, but it's a relic of optics. I'd probably rather test against the solidity than javascript however-- right now there are 2 sources of truth where the domain hashes are generated (solidity & rust), and testing with javascript would make 3 places (soldity & rust & javascript)